internal/rootcoord/alter_collection_task.go

```diff
@@ -48,8 +49,12 @@ func (a *alterCollectionTask) Prepare(ctx context.Context) error {
 
 func (a *alterCollectionTask) Execute(ctx context.Context) error {
 	// Now we only support alter properties of collection
-	if a.Req.GetProperties() == nil {
-		return errors.New("only support alter collection properties, but collection properties is empty")
+	if a.Req.GetProperties() == nil && a.Req.GetDeleteKeys() == nil {
+		return errors.New("only support alter collection properties, but collection properties or deletekeys is empty")
```

Comment:
The collection properties to alter and keys to delete must not be empty at the same time.